Dockerfile

```diff
@@ -1,21 +1,30 @@
-from golang:1.11 as build
+from golang:1.12 as build
 
 ENV SRC github.com/segmentio/aws-okta
 ARG VERSION
 
-WORKDIR /build
-COPY . /go/src/${SRC}
-
 RUN apt-get update
 RUN apt-get install --no-install-recommends -y \
     libusb-1.0-0-dev \
     ca-certificates \
     build-essential \
     git
 
-RUN CGO_ENABLED=1 go build -o aws-okta -ldflags="-X main.version=$VERSION" ${SRC}/cmd
+WORKDIR /src
+COPY . .
+
+RUN go mod download 
+RUN CGO_ENABLED=1 GOOS=linux go build -o /aws-okta -ldflags="-X main.version=$VERSION"
+
+FROM golang:1.12 as debug
+COPY --from=build /aws-okta /usr/local/bin/aws-okta
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y \
+    libusb-1.0-0 \
+    ca-certificates
+ENTRYPOINT ["/usr/local/bin/aws-okta"]
 
-FROM scratch
-COPY --from=build /build/aws-okta /
+FROM scratch as production
+COPY --from=build /aws-okta /aws-okta
```

Comment:
do we also want to copy this to `/usr/local/bin`?

Reply:
Sorry, there's a bunch of semi-related Dockerfile stuff in here that's distracting. For a scratch container, you don't need `/usr/local/bin`, but for a container you're going to exec into, that's based on Debian, you would.